fegrow/mlp.py

```diff
@@ -0,0 +1,175 @@
+"""Functionality for working with ML force fields."""
+
+import atexit
+import os
+import tempfile
+import urllib.request
+from abc import ABC, abstractmethod
+from typing import Literal
+
+from openff.toolkit import Molecule
+from openmmml import MLPotential as OMMMLPotential
+
+__all__ = [
+    "AVAILABLE_ML_FORCE_FIELDS",
+    "ANI2X",
+    "MACEOFF23Small",
+    "MACEOFF23Medium",
+    "MACEOFF23Large",
+    "EGRET1",
+]
+
+# Model accessed 24/05/25
+_EGRET1_MODEL_URL = (
+    "https://github.com/rowansci/egret-public/raw/227d6641e6851"
+    "eb1037d48712462e4ce61c1518f/compiled_models/EGRET_1.model"
+)
+
+AVAILABLE_ML_FORCE_FIELDS = Literal[
```

Comment:
Literal is not very extendable, but I guess for now we're not autodetecting available MLP classes